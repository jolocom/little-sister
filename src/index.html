<!DOCTYPE html>

<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport"
          content="width=device-width, initial-scale=1, user-scalable=no">

    <title>Little Sister</title>

    <link href="https://fonts.googleapis.com/css?family=Roboto:100,400,300,600"
          rel="stylesheet"
          type="text/css" />
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
          rel="stylesheet">

    <style type="text/css">
      * {
        margin: 0;
        padding: 0;
      }
      html, body {
        height: 100%;
      }
      body {
        font-family: Roboto;
      }
      #app {
        width: 100%;
        height: 100%;
      }
    </style>

    <!-- <meta http-equiv="Content-Security-Policy" content="default-src * data: gap: https://ssl.gstatic.com 'unsafe-eval'; style-src * 'unsafe-inline'; media-src *; img-src * data:"> -->
    <meta name="format-detection" content="telephone=no">
    <meta name="msapplication-tap-highlight" content="no">
    <meta name="viewport" content="user-scalable=no, initial-scale=1, maximum-scale=1, minimum-scale=1, width=device-width">
  </head>
  <body>
    <div id="app"></div>
    <script src="https://solid.github.io/releases/rdflib.js/rdflib-0.7.0.min.js"></script>
    <script src="https://d3js.org/d3.v4.min.js"></script>
    <script src="js/bundle.js"></script>
  </body>
</html>
