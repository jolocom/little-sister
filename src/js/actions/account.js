import Reflux from 'reflux'

let Actions = Reflux.createActions({
  login: {asyncResult: true},
  logout: {asyncResult: false},
  signup: {asyncResult: true}
})

export default Actions
