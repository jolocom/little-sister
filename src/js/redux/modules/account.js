import _ from 'lodash'
import Immutable from 'immutable'
import AccountsAgent from 'lib/agents/accounts'
import WebIDAgent from 'lib/agents/webid'
import { action, asyncAction } from './'
import { showMessage } from './snack-bar'

export const doLogin = asyncAction('account/login', 'doLogin', {
  expectedParams: ['username', 'password', 'updateUserEmail'],
  creator: (params) => {
    const wia = new WebIDAgent()
    const webId = wia.getWebId()

    // The user is already logged in.
    if (webId) {
      return async dispatch => {
        const accounts = new AccountsAgent()
        const loggedIn = await accounts.checkLogin(webId)
          .then(() => true).catch(() => false)

        if (loggedIn) {
          dispatch(doLogin.buildAction(params, async () => {
            return {
              username: localStorage.getItem('jolocom.username'),
              webId
            }
          }))
        } else {
          dispatch(doLogout())
        }
      }
    } else if (params.username && params.password) {
      return doLogin.buildAction(params, async () => {
        const accounts = new AccountsAgent()
        let promise
        if (params.updateUserEmail) {
          // init after activation only
          promise = accounts.loginAndSetup(
            params.username, params.password,
            params.updateUserEmail
          )
        } else {
          promise = accounts.login(params.username, params.password)
        }
        const account = await promise

        _saveAuthInfo(_saveToLocalStorage, params.username, account.webId)

        return {username: params.username, webId: account.webid}
      })
    } else {
      return dispatch => {} // Do nothing
    }
  }
})
export const doSignup = action('account', 'doSignup', {
  expectedParams: ['username', 'password', 'email', 'name'],
  creator: ({username, password, email, name}) => {
    return dispatch => {
      localStorage.setItem('jolocom.auth-mode', 'proxy')

      const accounts = new AccountsAgent()
      accounts.register(
        username, password, email, name
      ).then((account) => {
        dispatch(showEmailVerifyScreen())
      }).catch((e) => {
        if (e.message) {
          dispatch(showMessage({message: e.message}))
        } else {
          dispatch(showMessage({message: 'An account error has occured.'}))
        }
      })
    }
  }
})
export const doLogout = action('account', 'doLogout', {
  expectedParams: [],
  creator: params => {
    const authMode = localStorage.getItem('jolocom.auth-mode')

    const accounts = new AccountsAgent()
    if (authMode === 'proxy') {
      accounts.logout()
    }

    localStorage.removeItem('jolocom.username')
    localStorage.removeItem('jolocom.webId')
    localStorage.removeItem('jolocom.auth-mode')

    return doLogout.buildAction(params)
  }
})
export const doForgotPassword = asyncAction(
  'account', 'doForgotPassword',
  {
    expectedParams: ['username'],
    creator: params => {
      return dispatch => {
        const accounts = new AccountsAgent()
        return doForgotPassword.buildAction(
          params,
          accounts.forgotPassword(params.username)
            .then(() => dispatch(showMessage({
              message: 'An email was sent to you with further instructions.'
            })))
            .catch(e => {
              dispatch(showMessage({
                message: 'An error occured : ' + e
              }))
              throw e
            })
        )
      }
    }
  }
)
export const doResetPassword = asyncAction(
  'account', 'doResetPassword',
  {
    expectedParams: ['username', 'token', 'password'],
    creator: params => {
      return dispatch => {
        const accounts = new AccountsAgent()
        return doResetPassword.buildAction(
          params,
          accounts.resetPassword(params.username, params.token, params.password)
            .then(() => dispatch(showMessage({
              message: 'You can now log in with your new password.'
            })))
            .catch(e => {
              dispatch(showMessage({
                message: 'An error occured : ' + e
              }))
              throw e
            })
        )
      }
    }
  }
)
export const doActivateEmail = asyncAction(
  'account', 'doActivateEmail',
  {
    expectedParams: ['username', 'code'],
    creator: params => {
      return dispatch => {
        const accounts = new AccountsAgent()
        return doActivateEmail.buildAction(
          params,
          accounts.verifyEmail(params.username, params.code)
            .then(() => dispatch(showMessage({
              message: 'Your account has been activated!'
            })))
            .catch(e => {
              dispatch(showMessage({
                message: 'Account activation failed.'
              }))
              throw e
            })
        )
      }
    }
  }
)
export const doUpdateUserEmail = action(
  'account', 'doUpdateUserEmail',
  {
    expectedParams: ['email', 'webId', 'username'],
    creator: params => {
      return async dispatch => {
        const accounts = new AccountsAgent()
        await accounts.updateEmail(params.webId, params.email)

        _saveAuthInfo(_saveToLocalStorage, params.username, params.webId)
        dispatch({type: doLogin.id_success, result: {
          username: params.username,
          webId: params.webId
        }})
      }
    }
  }
)

export const showEmailVerifyScreen = action(
  'account/login', 'showEmailVerifyScreen',
  {expectedParams: []}
)
export const setUsername = action('account/login', 'setUsername', {
  expectedParams: ['username']
})
export const setPassword = action('account/login', 'setPassword', {
  expectedParams: ['password']
})

const initialState = Immutable.fromJS({
  username: '',
  password: '',
  userExists: false,
  emailVerifyScreen: false,
  emailVerifyCompleted: false,
  emailUpdateQueued: false,
  emailToBeInserted: '',
  webId: null
})

export default function reducer(state = initialState, action = {}) {
  state = state.set('login', loginReducer(state.get('login'), action))
  // state = state.set('login', true)

  switch (action.type) {
    case doLogin.id:
      return state.merge({loggingIn: true})
    case doLogin.id_success:
      return state.merge({
        loggingIn: false,
        username: action.result.username,
        webId: action.result.webId
      })
    case doLogout.id:
      return state.merge({
        loggingIn: false,
        username: null
      })
    case doActivateEmail.id_success:
      return state.merge({
        emailUpdateQueued: true,
        emailToBeInserted: action.result.email,
        emailVerifyCompleted: true
      })
    default:
      return state
  }
}

export function _saveToLocalStorage(data) {
  _.forEach(data, (v, k) => {
    localStorage.setItem(k, v)
  })
}

export function _saveAuthInfo(save, username, webId) {
  save({
    'jolocom.auth-mode': 'proxy',
    'jolocom.username': username,
    'jolocom.webId': webId
  })
}

// Login

export const setLoginUsername = action('account/login', 'setUsername', {
  expectedParams: ['username']
})
export const setLoginPassword = action('account/login', 'setPassword', {
  expectedParams: ['password']
})
export const setLoginErrorMessage = action('account/login', 'setErrorMessage', {
  expectedParams: ['field', 'message']
})
export const validateForm = action('account/login', 'validateForm', {
  creator: (params) => {
    return dispatch => {
      if (!params.username || !params.password) {
        if (!params.username) {
          dispatch(setLoginErrorMessage({
            field: 'user',
            message: 'Please enter a username'
          }))
        }
        if (!params.password) {
          dispatch(setLoginErrorMessage({
            field: 'pw',
            message: 'Please enter a password'
          }))
        }
      }
    }
  }
})

const initialLoginState = Immutable.fromJS({
  username: '',
  password: '',
  userErrorMsg: '',
  pwErrorMsg: '',
  failureMsg: ''
})

export function loginReducer(state = initialLoginState, action = {}) {
  switch (action.type) {
    case doLogin.id:
      return state.merge({
        failureMsg: ''
      })
    case doLogin.id_fail:
      return state.merge({
        failureMsg: 'Failed to log in'
      })
    case doLogin.id_success:
      return initialLoginState
    case setLoginUsername.id:
      return state.merge({
        userErrorMsg: '',
        username: action.username
      })
    case setLoginPassword.id:
      return state.merge({
        pwErrorMsg: '',
        password: action.password
      })
    case setLoginErrorMessage.id:
      return state.merge({
        [action.field + 'ErrorMsg']: action.message
      })
    case showEmailVerifyScreen.id:
      return state.merge({emailVerifyScreen: true})
    default:
      return state
  }
}
