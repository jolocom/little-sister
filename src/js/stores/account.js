import Reflux from 'reflux'
import Account from 'actions/account'
import {proxy} from 'settings'
import GraphAgent from 'lib/agents/graph'
import WebIdAgent from 'lib/agents/webid'
import rdf from 'rdflib'
import {PRED} from 'lib/namespaces'

import ConversationsActions from 'actions/conversations'

export default Reflux.createStore({
  listenables: Account,

  state: {
    loggingIn: true
  },

  getInitialState() {
    return this.state
  },

  onSignup(data) {
    localStorage.setItem('jolocom.auth-mode', 'proxy')

    let user = encodeURIComponent(data.username)
    let pass = encodeURIComponent(data.password)
    let name = data.name
    let email = data.email

    fetch(`${proxy}/register`, {
      method: 'POST',
      body: `username=${user}&password=${pass}`,
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
      }
    })
    .then((res) => {
      res.json().then((js) => {
        if (name || email) {
          let payload = {name, email}
          Account.login(data.username, data.password, payload)
        } else {
          Account.login(data.username, data.password)
        }
      })
    })
  },

  /* @summary in case the user specified a name / email when registering,
   * we update his already created profile with the data he introduced.
   * @param {string} name - the name we update the profile with.
   * @param {string} email - the email we update the profile with.
   */

  onSetNameEmail(webid, name, email) {
    let gAgent = new GraphAgent()
    let triples = []
    if (name) {
      triples.push({
        subject: rdf.sym(webid),
        predicate: PRED.givenName,
        object: name
      })
    }

    if (email) {
      triples.push({
        subject: rdf.sym(webid),
        predicate: PRED.email,
        // Keep an eye on this.
        object: rdf.sym(`mailto:${email}`)
      })
    }

    return gAgent.writeTriples(webid, triples, false)
  },

  /* @summary logs a user in. In case there's a updatePayload, first
   * applies the update and then logs the user.
   * @param {string} username
   * @param {string} password
   * @param {object} updatePayload - contains the name / email of a newly
   * created user. Only used when registering a new user.
  */

  onLogin(username, password, updatePayload) {
    const webId = localStorage.getItem('jolocom.webId')
    if (webId) {
      this.trigger({loggingIn: true})

      // Check if the cookie is still valid
      fetch(`${proxy}/proxy?url=${webId}`, {
        // using PATCH until HEAD is supported server-side; GET is too costly
        method: 'PATCH',
        credentials: 'include',
        headers: {
          'Content-Type': 'application/sparql-update'
        }
      }).then((res) => {
        if (!res.ok) {
          throw new Error(res.statusText)
        }
        Account.login.completed(
          localStorage.getItem('jolocom.username'),
          localStorage.getItem('jolocom.webId')
        )
      }).catch(() => {
        Account.logout()
      })
    } else if (username && password) {
      this.trigger({loggingIn: true})

      let user = encodeURIComponent(username)
      let pass = encodeURIComponent(password)

      fetch(`${proxy}/login`, {
        method: 'POST',
        body: `username=${user}&password=${pass}`,
        credentials: 'include',
        headers: {
          'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
        }
      }).then((res) => {
        res.json().then((js) => {
          if (updatePayload) {
            this.onSetNameEmail(
              js.webid, updatePayload.name, updatePayload.email
            ).then(() => {
              Account.login.completed(username, js.webid)
            }).then(() => {
              const webIdAgent = new WebIdAgent()
              webIdAgent.initInbox(js.webid)
            })
          } else {
            Account.login.completed(username, js.webid)
          }
        })
      })
    } else {
      this.trigger({loggingIn: false})
    }
  },

  onLoginCompleted(username, webId) {
    localStorage.setItem('jolocom.auth-mode', 'proxy')
    localStorage.setItem('jolocom.username', username)
    localStorage.setItem('jolocom.webId', webId)

    this.state = {
      loggingIn: false,
      username: username,
      webId: webId
    }
    
    // Load conversations for joining an existing one
    // when clicking on "chat" FAB on sbody's profile
    ConversationsActions.load(webId)

    this.trigger(this.state)
  },

  onLogout() {
    const authMode = localStorage.getItem('jolocom.auth-mode')

    if (authMode === 'proxy') {
      fetch(`${proxy}/logout`, {
        method: 'POST',
        credentials: 'include',
        // Not sure if the headers are necessary.
        headers: {
          'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
        }
      })
    }

    this.state = {
      loggingIn: false,
      username: null
    }

    localStorage.removeItem('jolocom.username')
    localStorage.removeItem('jolocom.webId')
    localStorage.removeItem('jolocom.auth-mode')

    this.trigger(this.state)
  },

  loggedIn() {
    // How would this work now?
    return localStorage.getItem('jolocom.webId')
  }
})
