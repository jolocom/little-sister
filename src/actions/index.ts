import * as accountActions from 'src/actions/account/'
import * as registrationActions from 'src/actions/registration/'
import * as navigationActions from 'src/actions/navigation/'
import * as genericActions from 'src/actions/generic/'
import * as ssoActions from 'src/actions/sso/'

export {
  accountActions,
  registrationActions,
  navigationActions,
  genericActions,
  ssoActions,
}
