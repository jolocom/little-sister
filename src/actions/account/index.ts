import { AnyAction, Dispatch } from 'redux'
import { genericActions, navigationActions } from 'src/actions/'
import { BackendMiddleware } from 'src/backendMiddleware'
import { routeList } from 'src/routeList'
import { DecoratedClaims, CategorizedClaims } from 'src/reducers/account'
import { SignedCredential } from 'jolocom-lib/js/credentials/signedCredential/signedCredential'
import {
  getClaimMetadataByCredentialType,
  getCredentialUiCategory,
  getUiCredentialTypeByType,
  instantiateIdentityWallet,
} from '../../lib/util'
import { cancelReceiving } from '../sso'

export const setDid = (did: string) => ({
  type: 'DID_SET',
  value: did,
})

export const setSelected = (claim: DecoratedClaims) => ({
  type: 'SET_SELECTED',
  selected: claim,
})

export const resetSelected = () => ({
  type: 'RESET_SELECTED',
})

export const handleClaimInput = (fieldValue: string, fieldName: string) => ({
  type: 'HANLDE_CLAIM_INPUT',
  fieldName,
  fieldValue,
})

export const toggleClaimsLoading = (value: boolean) => ({
  type: 'TOGGLE_CLAIMS_LOADING',
  value,
})

export const checkIdentityExists = () => async (
  dispatch: Dispatch<AnyAction>,
  getState: Function,
  backendMiddleware: BackendMiddleware,
) => {
  const { storageLib } = backendMiddleware

  try {
    const personas = await storageLib.get.persona()
    if (!personas.length) {
      dispatch(toggleLoading(false))
      return
    }

    dispatch(setDid(personas[0].did))
    await instantiateIdentityWallet(backendMiddleware)

    dispatch(toggleLoading(false))
    dispatch(navigationActions.navigatorReset({ routeName: routeList.Home }))
  } catch (err) {
    if (err.message.indexOf('no such table') === 0) {
      return
    }
    dispatch(genericActions.showErrorScreen(err))
  }
}

export const setIdentityWallet = () => async (
  dispatch: Dispatch<AnyAction>,
  getState: Function,
  backendMiddleware: BackendMiddleware,
) => {
  try {
    await instantiateIdentityWallet(backendMiddleware)
  } catch (err) {
    dispatch(genericActions.showErrorScreen(err))
  }
}

export const openClaimDetails = (claim: DecoratedClaims) => (
  dispatch: Dispatch<AnyAction>,
) => {
  dispatch(setSelected(claim))
  dispatch(
    navigationActions.navigate({
      routeName: routeList.ClaimDetails,
    }),
  )
}

export const saveClaim = () => async (
  dispatch: Dispatch<AnyAction>,
  getState: Function,
  backendMiddleware: BackendMiddleware,
) => {
  const { identityWallet, storageLib, keyChainLib } = backendMiddleware

  try {
    const did = getState().account.did.get('did')
    const claimsItem = getState().account.claims.toJS().selected
    const password = await keyChainLib.getPassword()

    const verifiableCredential = await identityWallet.create.signedCredential(
      {
        metadata: getClaimMetadataByCredentialType(claimsItem.credentialType),
        claim: claimsItem.claimData,
        subject: did,
      },
      password,
    )

    if (claimsItem.id) {
      await storageLib.delete.verifiableCredential(claimsItem.id)
    }

    await storageLib.store.verifiableCredential(verifiableCredential)
    await setClaimsForDid()

    dispatch(
      navigationActions.navigatorReset({
        routeName: routeList.Home,
      }),
    )
  } catch (err) {
    dispatch(genericActions.showErrorScreen(err))
  }
}

// TODO Currently only rendering  / adding one
export const saveExternalCredentials = () => async (
  dispatch: Dispatch<AnyAction>,
  getState: Function,
  backendMiddleware: BackendMiddleware,
) => {
  const { storageLib } = backendMiddleware
  const externalCredentials = getState().account.claims.toJS().pendingExternal
  const cred: SignedCredential = externalCredentials[0]

  if (cred.id) {
    await storageLib.delete.verifiableCredential(cred.id)
  }

  try {
    await storageLib.store.verifiableCredential(externalCredentials[0])
    dispatch(cancelReceiving())
  } catch (err) {
    dispatch(genericActions.showErrorScreen(err))
  }
}

export const toggleLoading = (value: boolean) => ({
  type: 'SET_LOADING',
  value,
})

export const setClaimsForDid = () => async (
  dispatch: Dispatch<AnyAction>,
  getState: Function,
  backendMiddleware: BackendMiddleware,
) => {
  dispatch(toggleClaimsLoading(true))
  const storageLib = backendMiddleware.storageLib

  const verifiableCredentials: SignedCredential[] = await storageLib.get.verifiableCredential()
  const claims = prepareClaimsForState(
    verifiableCredentials,
  ) as CategorizedClaims

  dispatch({
    type: 'SET_CLAIMS_FOR_DID',
    claims,
  })

  dispatch(toggleClaimsLoading(false))
}

const prepareClaimsForState = (credentials: SignedCredential[]) => {
  const categorizedClaims = {}
  const decoratedCredentials = convertToDecoratedClaim(credentials)

  decoratedCredentials.forEach(decoratedCred => {
    const uiCategory = getCredentialUiCategory(decoratedCred.credentialType)

    try {
      categorizedClaims[uiCategory].push(decoratedCred)
    } catch (err) {
      categorizedClaims[uiCategory] = [decoratedCred]
    }
  })

  return categorizedClaims
}

// TODO Util, make subject mandatory
export const convertToDecoratedClaim = (
  vCreds: SignedCredential[],
): DecoratedClaims[] =>
  vCreds.map(vCred => {
    const claimData = { ...vCred.claim }
    delete claimData.id

    return {
      credentialType: getUiCredentialTypeByType(vCred.type),
      claimData,
      id: vCred.id,
      issuer: vCred.issuer,
      subject: vCred.claim.id || 'Not found',
      expires: vCred.expires || undefined,
    }
  })
