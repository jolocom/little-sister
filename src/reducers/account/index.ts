import { combineReducers } from 'redux'
import { did } from 'src/reducers/account/did'
import { claims } from 'src/reducers/account/claims'
import { loading } from 'src/reducers/account/loading'
import { SignedCredential } from 'jolocom-lib/js/credentials/signedCredential/signedCredential'
import { ClaimEntry } from 'jolocom-lib/js/credentials/credential/types'

export interface DecoratedClaims {
  credentialType: string
  claimData: ClaimEntry
  id: string
  issuer: string
  subject: string
  expires?: Date
}

export interface CategorizedClaims {
  readonly [key: string]: DecoratedClaims[]
}

export interface ClaimsState {
  readonly loading: boolean
  readonly selected: DecoratedClaims
  readonly decoratedCredentials: CategorizedClaims
  readonly pendingExternal: SignedCredential[]
}

export interface DidState {
  readonly did: string
}

export interface LoadingState {
  readonly loading: boolean
}

// TODO avoid state.account.did.did access patterns
export interface AccountState {
  did: DidState
  claims: ClaimsState
  loading: LoadingState
}

export const accountReducer = combineReducers({
  did,
  claims,
  loading,
})
