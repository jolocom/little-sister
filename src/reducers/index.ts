import { combineReducers } from 'redux'
import { accountReducer, AccountState } from 'src/reducers/account/'
import {
  registrationReducer,
  RegistrationState,
} from 'src/reducers/registration/'
import { navigationReducer } from 'src/reducers/navigation/'
import { NavigationState } from 'react-navigation'
import { ssoReducer, SsoState } from 'src/reducers/sso/'

export const rootReducer = combineReducers({
  account: accountReducer,
  registration: registrationReducer,
  navigation: navigationReducer,
  sso: ssoReducer,
})

export interface RootState {
  readonly account: AccountState
  readonly registration: RegistrationState
  readonly navigation: NavigationState
  readonly sso: SsoState
}
