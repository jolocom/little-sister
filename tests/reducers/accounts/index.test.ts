import { accountReducer as reducer } from 'src/reducers/account/'
import { accountActions as actions } from 'src/actions/'

describe('account reducer', () => {
  it('should handle the DID_SET action ', () => {
    expect(reducer(undefined, actions.setDid('mock did'))).toMatchSnapshot()
  })

  it('should initialize correctly', () => {
    expect(reducer(undefined, { type: '@INIT' })).toMatchSnapshot()
  })
})
