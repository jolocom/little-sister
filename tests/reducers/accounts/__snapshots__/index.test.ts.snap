// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account reducer should handle the DID_SET action  1`] = `
Object {
  "claims": Immutable.Map {
    "loading": false,
    "selected": Immutable.Map {
      "credentialType": "",
      "claimData": Immutable.Map {},
      "id": "",
      "issuer": "",
      "subject": "",
    },
    "pendingExternal": Immutable.List [],
    "decoratedCredentials": Immutable.Map {
      "Personal": Immutable.List [
        Immutable.Map {
          "credentialType": "Name",
          "claimData": Immutable.Map {
            "givenName": "",
            "familyName": "",
          },
          "id": "",
          "issuer": "",
          "subject": "",
        },
      ],
      "Contact": Immutable.List [
        Immutable.Map {
          "credentialType": "Email",
          "claimData": Immutable.Map {
            "email": "",
          },
          "id": "",
          "issuer": "",
          "subject": "",
          "keyboardType": "email-address",
        },
        Immutable.Map {
          "credentialType": "Mobile Phone",
          "claimData": Immutable.Map {
            "telephone": "",
          },
          "id": "",
          "issuer": "",
          "subject": "",
          "keyboardType": "phone-pad",
        },
        Immutable.Map {
          "credentialType": "Postal Address",
          "claimData": Immutable.Map {
            "addressLine1": "",
            "addressLine2": "",
            "postalCode": "",
            "city": "",
            "country": "",
          },
          "id": "",
          "issuer": "",
          "subject": "",
        },
      ],
      "Other": Immutable.List [],
    },
  },
  "did": Immutable.Map {
    "did": "mock did",
  },
  "loading": Immutable.Map {
    "loading": true,
  },
}
`;

exports[`account reducer should initialize correctly 1`] = `
Object {
  "claims": Immutable.Map {
    "loading": false,
    "selected": Immutable.Map {
      "credentialType": "",
      "claimData": Immutable.Map {},
      "id": "",
      "issuer": "",
      "subject": "",
    },
    "pendingExternal": Immutable.List [],
    "decoratedCredentials": Immutable.Map {
      "Personal": Immutable.List [
        Immutable.Map {
          "credentialType": "Name",
          "claimData": Immutable.Map {
            "givenName": "",
            "familyName": "",
          },
          "id": "",
          "issuer": "",
          "subject": "",
        },
      ],
      "Contact": Immutable.List [
        Immutable.Map {
          "credentialType": "Email",
          "claimData": Immutable.Map {
            "email": "",
          },
          "id": "",
          "issuer": "",
          "subject": "",
          "keyboardType": "email-address",
        },
        Immutable.Map {
          "credentialType": "Mobile Phone",
          "claimData": Immutable.Map {
            "telephone": "",
          },
          "id": "",
          "issuer": "",
          "subject": "",
          "keyboardType": "phone-pad",
        },
        Immutable.Map {
          "credentialType": "Postal Address",
          "claimData": Immutable.Map {
            "addressLine1": "",
            "addressLine2": "",
            "postalCode": "",
            "city": "",
            "country": "",
          },
          "id": "",
          "issuer": "",
          "subject": "",
        },
      ],
      "Other": Immutable.List [],
    },
  },
  "did": Immutable.Map {
    "did": "",
  },
  "loading": Immutable.Map {
    "loading": true,
  },
}
`;
